eo-parser/src/main/java/org/eolang/parser/Syntax.java

```diff
@@ -110,41 +115,66 @@ public Syntax(
     /**
      * Compile it to XML and save.
      *
+     * <p>No exception will be thrown if the syntax is invalid. In any case, XMIR will
+     * be generated and saved. Read it in order to find the errors,
+     * at <tt>/program/errors</tt> XPath.</p>
+     *
      * @throws IOException If fails
+     * @checkstyle AnonInnerLengthCheck (100 lines)
      */
     public void parse() throws IOException {
         final List<Text> lines = this.lines();
-        final ANTLRErrorListener errors = new BaseErrorListener() {
+        final List<ParsingException> errors = new LinkedList<>();
+        final ANTLRErrorListener spy = new BaseErrorListener() {
             // @checkstyle ParameterNumberCheck (10 lines)
             @Override
             public void syntaxError(final Recognizer<?, ?> recognizer,
                 final Object symbol, final int line,
                 final int position, final String msg,
                 final RecognitionException error
             ) {
-                throw new ParsingException(
-                    String.format(
-                        "[%d:%d] %s: \"%s\"",
-                        line, position, msg,
-                        // @checkstyle AvoidInlineConditionalsCheck (1 line)
-                        lines.size() < line ? "EOF" : lines.get(line - 1)
-                    ),
-                    error,
-                    line
+                errors.add(
+                    new ParsingException(
+                        String.format(
+                            "[%d:%d] %s: \"%s\"",
+                            line, position, msg,
+                            // @checkstyle AvoidInlineConditionalsCheck (1 line)
+                            lines.size() < line ? "EOF" : lines.get(line - 1)
+                        ),
+                        error,
+                        line
+                    )
                 );
             }
         };
         final ProgramLexer lexer = new EoLexer(this.normalize());
         lexer.removeErrorListeners();
-        lexer.addErrorListener(errors);
+        lexer.addErrorListener(spy);
         final ProgramParser parser = new ProgramParser(
             new CommonTokenStream(lexer)
         );
         parser.removeErrorListeners();
-        parser.addErrorListener(errors);
+        parser.addErrorListener(spy);
         final XeListener xel = new XeListener(this.name, this.redundancy);
         new ParseTreeWalker().walk(xel, parser.program());
-        final XML dom = new XMLDocument(new Xembler(xel).domQuietly());
+        final XML dom = new XMLDocument(
+            new Xembler(
+                new Directives(xel).append(
+                    new org.cactoos.iterable.Joined<>(
```

Comment:
@yegor256 Do we need to use fully-qualified name of the class?

Reply:
@volodya-lombrozo yes, because we also import another `Joined` in this file